x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java

```diff
@@ -296,6 +298,38 @@ public void testParserBadActions() throws Exception {
         }
     }
 
+    public void testParserConsumesEntireDefinition() throws Exception {
+        WatchParser wp = createWatchparser();
+        String watchDef="{\n" +
+            "  \"trigger\": {\n" +
+            "    \"schedule\": {\n" +
+            "      \"interval\": \"10s\"\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"input\": {\n" +
+            "    \"simple\": {}\n" +
+            "  }},\n" + /* NOTICE EXTRA CLOSING CURLY BRACE */
+            "  \"condition\": {\n" +
+            "    \"script\": {\n" +
+            "      \"inline\": \"return false\"\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"actions\": {\n" +
+            "    \"logging\": {\n" +
+            "      \"logging\": {\n" +
+            "        \"text\": \"{{ctx.payload}}\"\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        try {
+            wp.parseWithSecrets("failure", false, new BytesArray(watchDef), new DateTime(), XContentType.JSON, true);
+            fail("This watch should fail to parse as there is an extra closing curly brace in the middle of the watch");
+        } catch (ElasticsearchParseException pe) {
+            assertThat(pe.getMessage().contains("unexpected data beyond"), is(true));
```

Comment:
This can use the `containsString` matcher instead of a boolean assert (the output on failure is much nicer).